pkg/network/usm/utils/debugger.go

```diff
@@ -46,49 +46,60 @@ type PathIdentifierWithSamplePath struct {
 	SamplePath string
 }
 
-// TracedProgramsEndpoint generates a summary of all active uprobe-based
-// programs along with their file paths and PIDs.
+// GetTracedProgramsEndpoint returns a callback for the given module name, that
+// generates a summary of all active uprobe-based programs along with their file paths and PIDs.
 // This is used for debugging purposes only.
-func TracedProgramsEndpoint(w http.ResponseWriter, _ *http.Request) {
-	otherutils.WriteAsJSON(w, debugger.GetTracedPrograms())
+func GetTracedProgramsEndpoint(moduleName string) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, _ *http.Request) {
+		otherutils.WriteAsJSON(w, debugger.GetTracedPrograms(moduleName))
+	}
 }
 
-// BlockedPathIDEndpoint generates a summary of all blocked uprobe-based
-// programs that are blocked in the registry along with their device and inode numbers, and sample path.
+// GetBlockedPathIDEndpoint returns a callback for the given module name, that
+// generates a summary of all blocked uprobe-based programs that are blocked in the
+// registry along with their device and inode numbers, and sample path.
 // This is used for debugging purposes only.
-func BlockedPathIDEndpoint(w http.ResponseWriter, _ *http.Request) {
-	otherutils.WriteAsJSON(w, debugger.GetAllBlockedPathIDs())
+func GetBlockedPathIDEndpoint(moduleName string) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, _ *http.Request) {
+		otherutils.WriteAsJSON(w, debugger.GetAllBlockedPathIDs(moduleName))
+	}
 }
 
-// ClearBlockedEndpoint clears the lists of blocked paths.
-func ClearBlockedEndpoint(_ http.ResponseWriter, _ *http.Request) {
-	debugger.ClearBlocked()
+// GetClearBlockedEndpoint returns a callback for the given module name, that clears the lists of blocked paths.
+func GetClearBlockedEndpoint(moduleName string) func(http.ResponseWriter, *http.Request) {
+	return func(http.ResponseWriter, *http.Request) {
+		debugger.ClearBlocked(moduleName)
+	}
 }
 
 var debugger *tlsDebugger
 
 type tlsDebugger struct {
 	mux        sync.Mutex
-	registries []*FileRegistry
-	attachers  map[string]Attacher
+	registries map[string][]*FileRegistry
+	attachers  map[string]map[string]Attacher
```

Comment:
What do you think about flattening the nested map? I feel that the current attachers struct is quite complex. Maybe we should consider adding a struct to improve the current map structure?

Reply:
added a syntactic sugar and documentation to make it easier